src/plugins/anki/anki-plugin.tsx

```diff
@@ -0,0 +1,291 @@
+
+  const onConfirm = () => {
+    saveConfiguration({
+      deckName: deckName,
+      deckId: deckId,
+    });
+  };
+
+  const { t } = useTranslation();
+
+  invoke("deckNames", 6).then((data: string[]) => {
+    setDecks(data);
+  });
+
+  const updateDeckId = (name: string) => {
+    setDeckName(name);
+    invoke("deckNamesAndIds", 6).then((data: any) => {
+      setDeckId(data[name]);
+    });
+  };
+
+  return (
+    <div className="AnkiWidget-config">
+      <div>{t("anki-plugin.name")}</div>
+
+      <div className="field">
+        <label>{t("anki-plugin.deck")}:</label>
+
+        <Select<string>
+          options={decks}
+          value={deckName}
+          onChange={updateDeckId}
+          getOptionKey={(o) => o}
+          getOptionLabel={(o) => o}
+        />
+      </div>
+
+      <Button className="save-config" onClick={onConfirm}>
+        Save
+      </Button>
+    </div>
+  );
+};
+
+const MainScreen = ({
+  config,
+  instanceId,
+}: WidgetRenderProps<AnkiPluginWidgetConfigType>) => {
+  const { t } = useTranslation();
+
+  const [currentCard, setCurrentCard] = useState<any>({});
+  const [cardsToLearn, setCardsToLearn] = useState<number[]>([]);
+  const [display, setDisplay] = useState("question");
+  const [reachable, setReachable] = useState(true);
+
+  const getCardAndSet = (card: number) => {
+    console.log(card);
+    invoke("cardsInfo", 6, {
+      cards: [card],
+    }).then((data: any[]) => {
+      const d = data[0];
+
+      d.question = { __html: d.question };
+      d.answer = { __html: d.answer };
+
+      console.log(d);
+      setCurrentCard(d);
+    });
+  };
+
+  const init = () => {
+    invoke("findCards", 6, {
+      query: '"deck:' + config?.deckName + '" is:due',
+    })
+      .catch((e) => {
+        setReachable(false);
+      })
+      .then((data: number[]) => {
+        if (reachable) {
+          setCardsToLearn(data);
+          getCardAndSet(data[0]);
+        } else {
+          setCardsToLearn([]);
+        }
+      });
+  };
+
+  useEffect(init, []);
+
+  const again = () => {
+    invoke("answerCards", 6, {
+      answers: [
+        {
+          cardId: cardsToLearn[0],
+          ease: 1,
+        },
+      ],
+    }).then((data) => {
+      if (cardsToLearn.length == 1) {
+        init();
+        return;
+      }
+      getCardAndSet(cardsToLearn[1]);
+      setCardsToLearn(cardsToLearn.slice(1));
+      setDisplay("question");
+    });
+  };
+
+  const easy = () => {
+    invoke("answerCards", 6, {
+      answers: [
+        {
+          cardId: cardsToLearn[0],
+          ease: 4,
+        },
+      ],
+    }).then((data) => {
+      if (cardsToLearn.length == 1) {
+        init();
+        return;
+      }
+      getCardAndSet(cardsToLearn[1]);
+      setCardsToLearn(cardsToLearn.slice(1));
+      setDisplay("question");
+    });
+  };
+
+  return (
+    <div className="AnkiWidget">
+      <p
+        className="setNameLine"
```

Comment:
Prefer kebab-case for simple classes (`AnkiWidget` above is a component root element, so it should remain in PascalCase)